components/playlist/browser/playlist_data_source.cc

```diff
@@ -77,10 +78,16 @@ void PlaylistDataSource::StartDataRequest(
       return;
     }
   } else if (type_string == "media") {
-    if (!service_->GetMediaPath(id, &data_path)) {
```

Comment:
GetMediaPath() returns not the actual file path, but a candidate path using the source URL. This made the player unable to play from local file. Fixed by using the actual file path.